railties/lib/rails/generators/rails/app/templates/bin/yarn.tt

```diff
@@ -1,7 +1,8 @@
 APP_ROOT = File.expand_path('..', __dir__)
 Dir.chdir(APP_ROOT) do
   begin
-    exec %w(yarnpkg) + ARGV
+    command = %w(yarnpkg) + ARGV
+    exec(*command)
```

Comment:
How about: `exec "yarnpkg", *ARGV`

Reply:
@justjake Please check the above comment?